documentation/jetty-documentation/src/main/asciidoc/operations-guide/jaspi/chapter.adoc

```diff
@@ -0,0 +1,45 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+[[og-jaspi]]
+=== JASPI
+
+Enabling this module allows Jetty to utilize authentication modules that implement the Jakara Authentication (JASPI) specification. JASPI provides an SPI (Service Provider Interface) for pluggable, portable, and standardized authentication modules. Compatible modules are portable between servers that support the JASPI specification. This module provides a bridge from Jakarta Authentication to the Jetty Security framework.  
+
+Only modules conforming to the "Servlet Container Profile" with the ServerAuthModule interface within the https://jakarta.ee/specifications/authentication/2.0/jakarta-authentication-spec-2.0.pdf[JakartaAuthentication] are supported her. These modules must be configured before start-up.  Operations for runtime registering or de-registering authentication modules are not supported. 
+
+[[og-jaspi-configuration]]
+==== Configuration
+
+[[og-jaspi-module]]
+===== The `jaspi` module
+
+Enable the `jaspi` module:
+
+----
+include::{JETTY_HOME}/modules/jaspi.mod[]
+----
+
+[[og-jaspi-xml]]
+===== Configure Jaspi
+
+When you enable the `jaspi` module, the example configuration file `jaspi-authmoduleconfig.xml` is copied into your `${jetty.base}/etc` directory:
+
+----
+include::{JETTY_HOME}/modules/jaspi/jaspi-authmoduleconfig.xml[tags=documentation]
+----
+
+It illustrates setting up HTTP Basic Authentication using a Jakarta Authentication module that comes packaged with jetty: `org.eclipse.jetty.security.jaspi.modules.BasicAuthenticationAuthModule`, and applies it for a context named `/ctx`. Other custom or 3rd party modules that are compatible with the `ServerAuthModule` interface in JASPI may be imported here.
+
+
+
```

Comment:
You need to add a description of how to finish the configuration setup for a webapp, which involves editing the context xml file for the webapp, and adding this line:  
```xml
  <Get name="securityHandler">
      <Set name="authenticatorFactory"><New class="org.eclipse.jetty.security.jaspi.JaspiAuthenticatorFactory"/></Set>
  </Get>
```